addons/product/tests/test_product_pricelist.py

```diff
@@ -351,3 +351,9 @@ def test_no_negative_cost(self):
         form = Form(self.product.product_tmpl_id)
         with self.assertRaises(ValidationError):
             form.standard_price = -5
+
+    def test_pricelist_rule_linked_to_product_variant(self):
+        """Verify that pricelist rules assigned to a variant remain linked after write."""
+        self.computer_SC234.pricelist_rule_ids = self.customer_pricelist.item_ids[0]
+        self.computer_SC234.invalidate_recordset()
+        self.assertTrue(self.computer_SC234.pricelist_rule_ids)
```

Comment:
```suggestion
        self.assertEqual(len(self.computer_SC234.pricelist_rule_ids), 1)
        self.assertIn(self.computer_SC234, self.pricelist.item_ids.product_id)
```
Probably better to ensure only a single item gets created, as well as check whether an item with the `product_id` gets added to the pricelist.